src/components/PdfFetcher/index.js

```diff
@@ -0,0 +1,163 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Spinner from 'react-spinkit'
+import chunkedRequest from 'chunked-request'
+import { connect } from 'react-redux'
+import escapeRegExp from 'escape-string-regexp'
+import { isEmpty } from 'lodash/lang'
+import compose from 'lodash/fp/compose'
+import {translate} from 'react-i18next'
+
+import withFetcher from 'endpoints/withFetcher'
+import LOCATIONS_ENDPOINT from 'endpoints/location'
+import PageModel from 'endpoints/models/PageModel'
+import LocationModel from 'endpoints/models/LocationModel'
+import style from './index.css'
+import Error from 'components/Error'
+
+class PdfFetcher extends React.Component {
+  static propTypes = {
+    page: PropTypes.instanceOf(PageModel).isRequired,
+    locations: PropTypes.arrayOf(PropTypes.instanceOf(LocationModel)).isRequired,
+    location: PropTypes.string.isRequired,
+    language: PropTypes.string.isRequired
+  }
+
+  constructor (params) {
+    super(params)
+    this.state = {pdf: '', loading: false}
+  }
+
+  componentWillMount () {
+    this.fetchUrl()
+  }
+
+  addPageIdsRecursively (pageIds, parentPage) {
+    parentPage.children.forEach((page) => {
+      pageIds.push(page.numericId)
+      this.addPageIdsRecursively(pageIds, page)
+    })
+  }
+
+  getFont () {
+    // todo: This still doesn't work perfect for ar. Improve this in WEBAPP-84
+    switch (this.props.language) {
+      case 'ar':
+        return 'aefurat'
+      case 'fa':
+        return 'dejavusans'
+      default:
+        return ''
+    }
+  }
+
+  getLocationTitle (page) {
+    try {
+      const location = this.props.locations.find((location) => location.code === page.title)
+      return location.name
+    } catch (e) {
```

Comment:
What are we catching here? I think find(...) just returns null/undefined to check for this and return page.title. Also the console log should just warn in this case.

Reply:
Indeed. Done.